nemo_curator/classifiers/multilingual_domain.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from dataclasses import dataclass
+from typing import List, Optional
+
+os.environ["RAPIDS_NO_INITIALIZE"] = "1"
+from crossfit.backend.torch.hf.model import HFModel
+from transformers import AutoConfig, AutoTokenizer
+
+from nemo_curator.classifiers.base import (
+    DistributedDataClassifier,
+    HFDeberta,
+    _get_suggest_memory_for_classifier,
+    _run_classifier_helper,
+)
+from nemo_curator.datasets import DocumentDataset
+
+DOMAIN_IDENTIFIER = "TODO"
+
+
+@dataclass
+class MultilingualDomainModelConfig:
+    model: str = "microsoft/mdeberta-v3-base"
+    fc_dropout: float = 0.2
+    max_len: int = 512
+
+
+class MultilingualDomainModel(HFModel):
+    def __init__(
+        self,
+        config: MultilingualDomainModelConfig,
+        autocast: bool = False,
+        max_mem_gb: Optional[int] = None,
+    ):
+        self.config = config
+        self.autocast = autocast
+        if max_mem_gb is None:
+            max_mem_gb = _get_suggest_memory_for_classifier()
+
+        super().__init__(self.config.model, max_mem_gb=max_mem_gb)
+
+    def load_model(self, device: str = "cuda"):
+        model = HFDeberta.from_pretrained(DOMAIN_IDENTIFIER)
+        model.set_autocast(self.autocast)
+        model = model.to(device)
+        return model.eval()
+
+    def load_tokenizer(self):
+        return AutoTokenizer.from_pretrained(DOMAIN_IDENTIFIER)
+
+    def load_config(self):
+        return AutoConfig.from_pretrained(DOMAIN_IDENTIFIER)
+
+
+class MultilingualDomainClassifier(DistributedDataClassifier):
```

Comment:
It might be cool to just integrate this with our `DomainClassifier` instead of creating a whole separate class for the multilingual case, like doing a `multilingual` parameter or something. I plan to experiment more with this after the Hugging Face release.